cmds/dhclient/dhcp6client/packetSock.go

```diff
@@ -0,0 +1,159 @@
+package dhcp6client
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+	"fmt"
+
+	"github.com/mdlayher/dhcp6"
+	"golang.org/x/sys/unix"
+)
+
+type packetSock struct {
+	fd      int
+	ifindex int
+}
+
+var bcastMAC = []byte{255, 255, 255, 255, 255, 255}
+
+func NewPacketSock(ifindex int) (*packetSock, error) {
```

Comment:
I see now that you're trying to mimic https://github.com/d2g/dhcp4client/blob/master/inetsock.go for this. That works, too.